src/Sparrow/Binary/Bits.cs

```diff
@@ -267,6 +268,9 @@ private static int PowerOf2Internal(int v)
             v |= v >> 16;
             v++;
 
+            if (v < 0)
+                throw new ArgumentException($"Could not return next power of 2 because the resulting number exceeds return type int");
```

Comment:
Add the original number here, but move the exception to a separate method

Reply:
For that matter, can't we just do a static check on > 1GB?

Reply:
The only PowerOf2 method to have this check is the one that returns an int. should we still create a dedicated method knowing only one method is going to use it?

Reply:
I'm actually saying that you can do the check at the top, not the end.